dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/DataAPI.kt

```diff
@@ -19,78 +17,64 @@ import javax.validation.Valid
+        summary = "Retrieve list of existing data for this data type.",
+        description = "List is composed of meta information containing the data ID, the company ID and the data type."
     )
     @ApiResponses(
         value = [
-            ApiResponse(responseCode = "200", description = "Successfully retrieved list of data.")
+            ApiResponse(responseCode = "200", description = "Successfully retrieved list of entries.")
         ]
     )
     @GetMapping(
         value = ["/data"],
         produces = ["application/json"]
     )
     /**
-     * Returns the meta information (id and name) of all currently available data sets
+     * Returns the meta information (dataId, companyId and dataType) of available data sets
      */
     fun getData(): ResponseEntity<List<DataSetMetaInformation>>
 
     @Operation(
         summary = "Upload new data set.",
-        description = "The uploaded data set is added to the data store, the generated id is returned."
+        description = "The uploaded data is added to the data store, the generated data id is returned."
     )
     @ApiResponses(
         value = [
             ApiResponse(responseCode = "200", description = "Successfully added data set to the data store.")
         ]
     )
     @PostMapping(
-        value = ["/data"],
+        value = ["/{companyId}"],
         produces = ["application/json"],
         consumes = ["application/json"]
     )
     /**
      * A method to store a provided data set via dataland into the data store
+     * @param companyId the ID of the company the data belongs to
      * @param dataSet a set of data to be stored
-     * @return meta information of the stored data (id and name)
+     * @return the ID of the created entry in the data store
      */
-    fun postData(@Valid @RequestBody dataSet: DataSet): ResponseEntity<DataSetMetaInformation>
+    fun postData(@PathVariable("companyId") companyId: String, @Valid @RequestBody dataSet: T): ResponseEntity<String>
 
     @Operation(
         summary = "Retrieve specific data set from the data store.",
-        description = "The data set identified via the provided id is retrieved."
+        description = "The data set identified via the provided data ID is retrieved."
     )
     @ApiResponses(
         value = [
             ApiResponse(responseCode = "200", description = "Successfully retrieved data set.")
         ]
     )
     @GetMapping(
-        value = ["/data/{id}"],
-        produces = ["application/json"]
-    )
-    /**
-     * A method to retrieve a specific data set identified by its id
-     * @param id identifier used to uniquely determine the data set in the data store
-     * @return the complete data stored under the provided id
-     */
-    fun getDataSet(@PathVariable("id") id: String): ResponseEntity<DataSet>
-
-    @GetMapping(
-        value = ["/data/skyminder/{code}/{companyName}"],
+        value = ["/{dataId}"],
```

Comment:
Ich finde das verwirrend: 
POST auf `/17` wird Daten für die Company Nummer 17 hinzufügen.
GET auf `/17` wird Datensatz Nummer 17 zurückliefern - möglicherweise nicht mal vom Typ `T` und möglicherweise von einer anderen Company. 

Ich finde, die Path-Variable sollte nicht mit der Methode die Bedeutung verändern.

Können wir hier nochmal drüber nachdenken?

Reply:
Klingt nach einer Lösung.  Kann es so umsetzen.   @ppuetsch  was meinst du?

Reply:
ja - gern. Habe dazu mit @SiegfriedSobkowiak telefoniert.

Reply:
Um das umzusetzen, habe ich versucht ein neues Model "PostDataRequestBody" zu erstellen, und darin companyId + data set zu integrieren.
Mit diesem Model wollte ich dann den RequestBody für den POST Endpunkt /eutaxonomies definieren.

Problem: Im Model konnte ich den generischen Datentyp "T" nicht verwenden.   Was ist die Lösung?
Gerne morgen einmal besprechen.

Reply:
Umgesetzt von @SiegfriedSobkowiak  in eigenem Branch.   Muss noch in diesen Branch reingemerget werden.

Reply:
(Warten bzw. arbeiten noch auf grüne CI hin in dem anderen Branch)

Reply:
ok